### PR TITLE
[feat] implement parsing extension on root level openapi

### PR DESCRIPTION
**Pull request by ymohl-cl**
- Use VendorExtensions from spec.Swagger to parse the extension on root
level
- Provide tests
- Provide documentation usage

**Describe the PR**
add extensions on root level.

**Relation issue**
https://github.com/swaggo/swag/issues/398

**Additional context**
Thanks for read ;)

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -180,6 +194,32 @@ func TestParser_ParseGeneralApiInfoTemplated(t *testing.T) {
 	assert.Equal(t, expected, string(b))
 }
 
+func TestParser_ParseGeneralApiInfoExtensions(t *testing.T) {
+	// should be return an error because extension value is not a valid json
+	func() {
```
Comment (easonlin404) [inline parser_test.go]: why needs enclosed with `func()`?

**Comment by easonlin404** [comment]
@ymohl-cl Please help resolve conflicts.

File: parser_test.go (hunk 1 above)
Comment (ymohl-cl) [reply parser_test.go]: just a pattern to exclude side effect from other tests

**Comment by ymohl-cl** [comment]
done ;)

**Comment by easonlin404** [comment]
@ymohl-cl Thanks for your contributions. 💯

**Pull request merged**